guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc

```diff
@@ -0,0 +1,457 @@
+----
+$ hammer | less
+----
+[[sect-CLI_Guide-Authentication]]
+== Authentication
+
+A {Project} user must prove their identity to {ProjectName} when entering hammer commands.
+Hammer commands can be run manually or automatically.
+In either case, hammer requires {Project} credentials for authentication.
+There are three methods of hammer authentication:
+
+* Hammer authentication session
+* Storing credentials in the hammer configuration file
+* Providing credentials with each hammer command
+
+The hammer configuration file method is recommended when running commands automatically.
+For example, running {Project} maintenance commands from a cron job.
+When running commands manually, Red{nbsp}Hat recommends using the hammer authentication session and providing credentials with each command.
+
+[[sect-CLI_Guide-Authentication-Hammer_Authentication_Session]]
+=== Hammer authentication session
+
+The hammer authentication session is a cache that stores your credentials, and you have to provide them only once, at the beginning of the session.
+This method is suited to running several hammer commands in succession, for example a script containing hammer commands.
+In this scenario, you enter your {Project} credentials once, and the script runs as expected.
+By using the hammer authentication session, you avoid storing your credentials in the script itself and in the `~/.hammer/cli.modules.d/foreman.yml` hammer configuration file.
+
+See the instructions on how to use the sessions:
+
+* To enable sessions, add `:use_sessions: true` to the `~/.hammer/cli.modules.d/foreman.yml` file:
++
+----
+:foreman:
+ :use_sessions: true
+----
+Note that if you enable sessions, credentials stored in the configuration file will be ignored.
++
+* To start a session, enter the following command:
++
+----
+# hammer auth login
+----
++
+You are prompted for your {Project} credentials, and logged in.
+You will not be prompted for the credentials again until your session expires.
++
+* The default length of a session is 60 minutes.
+You can change the time to suit your preference.
+For example, to change it to 30 minutes, enter the following command:
++
+----
+# hammer settings set --name idle_timeout --value 30
+Setting [idle_timeout] updated to [30]
+----
++
+* To see the current status of the session, enter the following command:
++
+----
+# hammer auth status
+----
++
+* To end the session, enter the following command:
++
+----
+# hammer auth logout
+----
+
+[[sect-CLI_Guide-Authentication-Hammer_Configuration_File]]
+=== Hammer configuration file
+
+If you ran the {Project} installation with `--foreman-initial-admin-username` and `--foreman-initial-admin-password` options, credentials you entered are stored in the `~/.hammer/cli.modules.d/foreman.yml` configuration file, and hammer does not prompt for your credentials.
+
+You can also add your credentials to the `~/.hammer/cli.modules.d/foreman.yml` configuration file manually:
```

Comment:
Might be good to add that the installer won't update the file and if you changed your credentials then you need to do that manually. You already have the instructions.